chartpress.py

```diff
@@ -750,7 +807,43 @@ def build_chart(
 
     # decide a version string
     if version is None:
-        version = _get_identifier_from_paths(*paths, long=long)
+        if use_chart_version:
+            # avoid adding .git... to chart version multiple times!
+            base_version = _trim_version_suffix(chart["version"])
+            # check for default placeholder tags
+            # avoid making weird combination of `1.0.0-set.by.chartpress.git.1.habc`
+            if "set.by.chartpress" in chart["version"]:
+                raise ValueError(
+                    f"Chart {chart_file} has placeholder version {chart['version']}, with `useChartVersion: true`."
+                    f" Set the version in the {chart_file} to a base pre-release tag (your _next_ release),"
+                    " e.g. `2.0.0-0.dev` or `2.0.0-alpha.1` and run chartpress again."
+                )
+
+            # require starting from a prerelease tag, to ensure correct ordering
+            if "-" not in chart["version"]:
+                if reset:
+                    raise ValueError(
+                        "--reset after a release must also specify --tag $VERSION to reset to"
+                        " when using useChartVersion: true."
+                        " This should be your _next_ pre-release version,"
+                        " e.g. `chartpress --reset --tag 1.0.1-0.dev`"
+                    )
+                raise ValueError(
+                    f"Chart {chart_file} has non-prerelease version {chart['version']} with `useChartVersion: true`"
+                    f" Set the version in the {chart_file} to a base pre-release tag (your _next_ release),"
+                    " e.g. `2.0.0-0.dev` or `2.0.0-alpha.1` and run chartpress again."
+                )
+        else:
+            base_version = None
+        if reset:
+            # resetting, use the base version without '.git...'
+            version = base_version
+        else:
+            # derive the full version, with possible '.git...'
+            version = _get_identifier_from_paths(
+                *paths, long=long, base_version=base_version
+            )
```

Comment:
Hmmmm the `if reset:` clause above is evaluated only when `version is None`. But it seems that if `reset` is True, then `build_chart()` is always called with the `version` parameter set based on this, where we see the only call to `build_chart`.

```python
        if args.tag:
            # tag specified, use that version
            forced_version = args.tag
        elif args.reset and not use_chart_version:
            # resetting, get version from chartpress.yaml,
            # ignoring current version in Chart.yaml
            forced_version = chart.get("resetVersion", "0.0.1")

        if not args.list_images:
            # update Chart.yaml with a version
            chart_version = build_chart(
                chart["name"],
                paths=_get_all_chart_paths(chart),
                version=forced_version,
                long=args.long,
                reset=args.reset,
                strict_version=args.publish_chart,
                use_chart_version=use_chart_version,
            )
```

Leaving this in, even though its currently unreachable is also not great because if it ever becomes reachable, then version could end up as `None` after this code segment, and then finally the Chart.yaml etc could get a version field set to `None` as well.

Reply:
I think the following change is equivalent to what we have now, because of the unreachability of the `if` section.

```suggestion
        # derive the full version, with possible '.git...'
        version = _get_identifier_from_paths(
            *paths, long=long, base_version=base_version
        )
```

Reply:
This isn't unreachable, it's the path taken by `chartpress --reset` (e.g. as called by the pre-commit hook) when use_chart_version is True. I'll try to clear up the multiple paths, probably by moving the selection of the reset version nearby so they two paths aren't so different.

Reply:
I didn't understand that you are right about this, but reviewing the new changes makes me fine about the current state which seems good to me - thank you!